rails/app/controllers/api/v1/projects_controller.rb

```diff
@@ -12,4 +12,15 @@ def index
     render :json => result
   end
 
+  def show
+    project = Admin::Project
+                   .where(id: params[:id])
+                   .first
```

Comment:
Good point, I'd forgotten about the error handling below.

The most standard Rails approach is to use the `find` above and then add a 
`rescue_from ActiveRecord::RecordNotFound, :with => :[some function]`
to handle the case where the record is not found.

An example of `rescue_from` in the API is here: https://github.com/concord-consortium/rigse/blob/1044a124297484b570863ce8586eb5c1d1b362d1/rails/app/controllers/api/api_controller.rb#L3

The non API handling of these 'not found' errors is setup here:
https://github.com/concord-consortium/rigse/blob/1044a124297484b570863ce8586eb5c1d1b362d1/rails/app/controllers/application_controller.rb#L48

That non-api method can't be used. because it returns a html page instead of JSON.

Reply:
Are the changes I made for this good?

Reply:
The api_controller already handles the ParameterMissing so you don't need to add that to this controller.

I'd move 'rescue_from ActiveRecord::RecordNotFound, :with => :record_not_found` into the api controller since this will be generally useful.